content/03.methods.md

```diff
@@ -5,8 +5,72 @@ Talk about dataset - Pubtator
 Talk about preprocessing Pubtator
 Talk about hand annotations for each realtion
 
-## Label Functions
-describe what a label function is and how many we created for each relation
+## Label Functions for Annotating Sentences
+In natural language processing a common problem is having too much data and not enough ground truth annotations.
+Data programming [@arxiv:1605.07723] circumvents this issue by quickly annotating large datasets by using multiple noisy signals emitted by label functions.
+Label functions are simple pythonic functions emit: a positive label (1), a negative label (-1) or abstain from emitting a label (0).
+We combine these functions using a generative model to output a single annotation, then use a discriminator model to improve upon this model.
+Label functions fall into these three categories: databases, text patterns and domain heuristics.
+These categories are described below using the following candidate sentence as an example: \"<span style="color:#02b3e4">PTK6</span> may be a novel therapeutic target for <span style="color:#875442">pancreatic cancer</span>."
+
+**Databases**: These label functions incorporate existing databases to generate a signal, as seen in distant supervision [@doi:10.3115/1690219.1690287].
+These functions assert if a candidate sentence's co-mention pair is present in a given database.
+If the pair is present, emit a positive label and abstain otherwise.
+If the pair isn't present in any existing database, then a separate label function will emit a negative label.
+We use a separate label function to prevent the label imbalance problem, which is less studied candidates will be drowned out by negative labels.
+These negative labels increase the likelihood of misclassification when training the generative model.
+
+$$ \Lambda_{DB}(\color{#875442}{D}, \color{#02b3e4}{G}) = 
+\begin{cases}
+ 1 & (\color{#875442}{D}, \color{#02b3e4}{G}) \in DB \\
+0 & otherwise \\
+\end{cases} $$
+
+$$ \Lambda_{\neg DB}(\color{#875442}{D}, \color{#02b3e4}{G}) = 
+\begin{cases}
+ -1 & (\color{#875442}{D}, \color{#02b3e4}{G}) \notin DB \\
+0 & otherwise \\
+\end{cases} $$
+
+**Text Patterns**: These label functions are designed to use keywords and sentence context to generate a signal. 
+For example, a label function could focus on the number of words between two mentions or focus on the grammatical structure of a sentence.
+These functions emit a positive or negative output depending on the situation.
+In general, those focused on keywords emit positives and those focused on negation emit negatives.
+
+$$ \Lambda_{TP}(\color{#875442}{D}, \color{#02b3e4}{G}) = 
+\begin{cases}
+ 1 & "target" \> \in Candidate \> Sentence \\
+ 0 & otherwise \\
+\end{cases} $$
+
+$$ \Lambda_{TP}(\color{#875442}{D}, \color{#02b3e4}{G}) = 
+\begin{cases}
+ -1 & 	`"VB" \> \notin pos\_tags(Candidate \> Sentence) \\
```

Comment:
Does this backtick "`" ever end? The syntax highlighting in github says it doesn't